clients/src/test/java/org/apache/kafka/clients/producer/internals/BufferPoolTest.java

```diff
@@ -416,6 +418,10 @@ public Void call() throws Exception {
             executor.submit(work);
         }
 
+        // Wait for the workers to be blocked in their buffer allocations.
+        ready.await(15, TimeUnit.SECONDS);
+        Thread.sleep(100);
```

Comment:
Do you know why the sleep is needed ?

From my experiment, 1 millis sleep suffices.

The timeout for completed.await() below can be shortened as well since the allocate() call would return immediately.

Reply:
It's not necessary, but was done to ensure better predictability.

In the latch implementation, it may notify a futex which, depending on the scheduler, may pause the worker thread and schedule back to the main thread which would call close() before the worker was waiting in the buffer pool. Adding a sleep deschedules the main thread again so that the worker may continue.

1 ms should accomplish the same thing. As @hachikuji noted in the previous PR, timeouts should be fairly lax so that they don't induce flakiness when running on Jenkins, so I'll defer to him on any modifications there.